src/common/tvgMath.h

```diff
@@ -393,6 +393,50 @@ static inline void log(const Point& pt)
 }
 
 
+static inline bool closed(const Point& lhs, const Point& rhs, float tolerance)
+{
+    float dx = lhs.x - rhs.x;
+    float dy = lhs.y - rhs.y;
+    return (dx * dx + dy * dy) < (tolerance * tolerance);
+}
+
+
+static inline void point2Line(const Point& point, const Point& start, const Point& vec, const float vecLenSq, float& maxDist, float& minT, float& maxT)
+{
+    Point offset = point - start;
+    auto area = cross(vec, offset);
+    auto dist = fabsf(area) / sqrtf(vecLenSq); // if lineVecLenSq == 0, mean closed() hasn't been called
+    if (dist > maxDist) maxDist = dist;
+
+    auto t = dot(offset, vec) / vecLenSq;
+    if (t < minT) minT = t;
+    if (t > maxT) maxT = t;
+}
+
+
+static inline void validateCubic(const Point& start, const Point& ctrl1, const Point& ctrl2, const Point& end, float& maxDist, float& minT, float& maxT)
```

Comment:
Please inline `validateCubic()` and `point2Line()` into optimize() using closure style.